aggregator_core/src/datastore.rs

```diff
@@ -1204,6 +1204,87 @@ impl<C: Clock> Transaction<'_, C> {
+    /// will not stop additional aggregation jobs from being created later with different
+    /// aggregation parameters.
+    ///
+    /// This should only be used with VDAFs with a non-unit type aggregation parameter. If a VDAF
+    /// has the unit type as its aggregation parameter, then
+    /// `get_unaggregated_client_report_ids_for_task` should be used instead. In such cases, it is
+    /// not necessary to wait for a collection job to arrive before preparing reports.
+    ///
+    /// This function deliberately ignores the `client_reports.aggregation_started` column, which
+    /// only has meaning for VDAFs without aggregation parameters.
+    #[tracing::instrument(skip(self), err)]
+    pub async fn get_unaggregated_client_report_ids_by_collect_for_task<const SEED_SIZE: usize, A>(
+        &self,
+        task_id: &TaskId,
+        limit: usize,
+    ) -> Result<Vec<(A::AggregationParam, ReportMetadata)>, Error>
+    where
+        A: vdaf::Aggregator<SEED_SIZE, 16> + VdafHasAggregationParameter,
+    {
+        // TODO(#224): lock retrieved client_reports rows
+        // TODO(#225): use get_task_primary_key_and_expiry_threshold as in
+        // get_unaggregated_client_reports_for_task
+        let stmt = self
+            .prepare_cached(
+                "WITH unaggregated_client_report_ids AS (
+                    SELECT DISTINCT report_id, client_timestamp, collection_jobs.aggregation_param
+                    FROM collection_jobs
+                    INNER JOIN client_reports
+                    ON collection_jobs.task_id = client_reports.task_id
+                    AND client_reports.client_timestamp <@ collection_jobs.batch_interval
+                    LEFT JOIN (
+                        SELECT report_aggregations.id, report_aggregations.client_report_id,
+                            aggregation_jobs.aggregation_param
+                        FROM report_aggregations
+                        INNER JOIN aggregation_jobs
+                        ON aggregation_jobs.id = report_aggregations.aggregation_job_id
+                        WHERE aggregation_jobs.task_id = (SELECT id FROM tasks WHERE task_id = $1)
+                    ) AS report_aggs
+                    ON report_aggs.client_report_id = client_reports.report_id  -- this join is very inefficient, fix before deploying in non-test scenario
+                    AND report_aggs.aggregation_param = collection_jobs.aggregation_param
+                    WHERE client_reports.task_id = (SELECT id FROM tasks WHERE task_id = $1)
+                    AND collection_jobs.task_id = (SELECT id FROM tasks WHERE task_id = $1)
+                    AND collection_jobs.state = 'START'
+                    AND report_aggs.id IS NULL
+                    LIMIT $2::BIGINT
+                ),
+                updated_client_reports AS (
+                    UPDATE client_reports SET aggregation_started = TRUE
```

Comment:
Q: is there a functional need to set these as `aggregation_started`? or is this bookkeeping?

Reply:
No, this just got resurrected from #1508, but it doesn't do anything anymore. I simplified the query considerably.

Reply:
I spoke too soon: the integration test fails if we don't set `aggregation-started = TRUE`. I believe I'll need to revisit this as I continue in this area (because in the presence of varying aggregation parameters, a single boolean doesn't suffice) but for now I'll leave the query as-is.